[EDU-3244] add cache settings command ref (#435)
aziontech/docs: src/content/docs/en/pages/devtools/cli/azion-cli-alpha/cache-settings/cache-settings.mdx

---

## Create

### Usage

```bash
$ azion create cache-setting --application-id 1673635839 --name "phototypesetting"
```

### Required flags

#### application-id

The `--application-id` flag sets the unique identifier for the edge application.

#### name

The `--name` flag sets the name of the Cache Settings configuration being created.

### Optional flags

#### adaptive-delivery-action

The `--adaptive-delivery-action` option informs the Cache Settings configuration adaptive delivery action. The default value is `ignore`.

#### browser-cache-settings

The `--browser-cache-settings` option configures the amount of time that the content is cached in the web browser. The default value is `honor`.

#### browser-cache-settings-maximum-ttl

The `--browser-cache-settings-maximum-ttl` option informs Browser Cache Settings configuration maximum TTL.

#### cache-by-cookies

The `--cache-by-cookies` option defines if cache by cookies is active. The default value is `ignore`.

#### cache-by-query-string

The `--cache-by-query-string` option defines how the content should be cached according to variations of the Query String in URLs. The default value is `ignore`.

#### cdn-cache-settings

> The `--cdn-cache-settings` option configures the amount of time Azion's Edge Applications take to cache the content. It can either Honor Origin Cache Headers or Override Cache Settings. The default value is `honor`.

Reviewer: ```suggestion
The `--cdn-cache-settings` option configures the amount of time Azion's edge applications take to cache the content. It can either Honor Origin Cache Headers or Override Cache Settings. The default value is `honor`.
```

#### cnd-cache-settings-maximum-ttl

> The `--cnd-cache-settings-maximum-ttl` option informs CDN Cache Settings configuration maximum TTL. The default value is 60.

Reviewer: ```suggestion
The `--cnd-cache-settings-maximum-ttl` option informs CDN Cache Settings configuration maximum TTL. The default value is `60`.
```

#### cookie-names

The `--cookie-names` option distinguishes objects in the Azion cache by the name/value of cookies.

#### enable-caching-for-options

The `--enable-caching-for-options` option defines if caching for options is active. The default value is `false`.

#### enable-caching-for-post

The `--enable-caching-for-post` option defines if caching for post is active.

#### enable-caching-string-sort

The `--enable-caching-string-sort` option defines if caching string sort is active.

#### file

> The `--file` option specifies the path to a JSON file containing the attributes of the Cache Settings configuration that will be created. You can use `-` for reading from stdin.

Reviewer 2: ```suggestion
The `--file` option specifies the path to a JSON file containing the attributes of the Cache Settings configuration that'll be created. You can use `-` for reading from stdin.
```

#### l2-caching-enabled

The `--l2-caching-enabled` option defines if L2 caching is active.

#### query-string-fields

The `--query-string-fields` option gives a list of query string parameters to be considered in the Cache Settings configuration, segregating the cache to the same URL.

#### slice-configuration-enabled

The `--slice-configuration-enabled` option defines if slice configuration is active.

#### slice-configuration-range

The `--slice-configuration-range` option informs the slice configuration range.

#### slice-l2-caching-enabled

The `--slice-l2-caching-enabled` option defines if slice L2 caching is active.

---

## List

### Usage

```bash
$ azion list cache-setting --application-id 16736354321
```

### Optional flags

#### application-id

The `--application-id` flag sets the unique identifier for the edge application.

#### details

The `--details` flag displays all relevant fields when listing.

#### filter

The `--filter` option filters items by their name.

#### help

The `--help` option displays more information about the `azion list cache-setting` command.

#### order-by

The `--order-by` option sorts the output based on the selected field.

#### page

> The `--page` option returns a page of the list according to its number. The default value is 1.

Reviewer: ```suggestion
The `--page` option returns a page of the list according to its number. The default value is `1`.
```

#### page-size

> The `--page-size` option defines how many items should be returned per page. The default value is 10.

Reviewer: ```suggestion
The `--page-size` option defines how many items should be returned per page. The default value is `10`.
```

#### sort

The `--sort` option defines the order of the items on the list. The value should be either `asc` or `desc`.

---

## Describe

### Usage

```bash
$ azion describe cache-setting --application-id 1673635839 --cache-setting-id 107313
```

### Required flags

#### application-id

The `--application-id` flag sets the unique identifier for the edge application.

#### cache-setting-id

The `--cache-setting-id` flag sets the unique identifier for a Cache Settings configuration.

### Optional flags

#### format

The `--format` option changes the output format. Pass the value `json` to the flag to get the output in JSON format.

#### help

The `--help` option displays more information about the `azion describe cache-setting` command.

#### out

> The `--out` option exports the output to the given file path or file name with extension. For example, `--out "./tmp/test.json"`.

Reviewer: ```suggestion
The `--out` option exports the output to the given file path or file name with an extension. Example: `--out "./tmp/test.json"`.
```

---

## Update

### Usage

```bash
$ azion update cache-setting --application-id 1673635839 --cache-setting-id 123123421 --name "phototypesetting"
```

### Required flags

#### application-id

The `--application-id` flag sets the unique identifier for the edge application.

#### cache-setting-id

The `--cache-setting-id` flag sets the unique identifier for a Cache Settings configuration.

#### name

The `--name` flag sets the name of the Cache Settings configuration being updated.

### Optional flags

#### adaptive-delivery-action

The `--adaptive-delivery-action` option informs the Cache Settings configuration adaptive delivery action. The default value is `ignore`.

#### browser-cache-settings

The `--browser-cache-settings` option configures the amount of time that the content is cached in the web browser. The default value is `honor`.

#### browser-cache-settings-maximum-ttl

The `--browser-cache-settings-maximum-ttl` option informs Browser Cache Settings configuration maximum TTL.

#### cache-by-cookies

The `--cache-by-cookies` option defines if cache by cookies is active. The default value is `ignore`.

#### cache-by-query-string

> The `--cache-by-query-string` option defines how the content should be cached according to variations of the Query String in URLs. The default value is `ignore`.

Reviewer 2: ```suggestion
The `--cache-by-query-string` option defines how the content should be cached according to variations of the query string in URLs. The default value is `ignore`.
```

#### cdn-cache-settings

> The `--cdn-cache-settings` option configures the amount of time Azion's Edge Applications take to cache the content. It can either Honor Origin Cache Headers or Override Cache Settings. The default value is `honor`.

Reviewer: ```suggestion
The `--cdn-cache-settings` option configures the amount of time Azion's edge applications take to cache the content. It can either Honor Origin Cache Headers or Override Cache Settings. The default value is `honor`.
```

#### cnd-cache-settings-maximum-ttl

> The `--cnd-cache-settings-maximum-ttl` option informs CDN Cache Settings configuration maximum TTL. The default value is 60.

Reviewer: ```suggestion
The `--cnd-cache-settings-maximum-ttl` option informs CDN Cache Settings configuration maximum TTL. The default value is `60`.
```

#### cookie-names

The `--cookie-names` option distinguishes objects in the Azion cache by the name/value of cookies.

#### enable-caching-for-options

The `--enable-caching-for-options` option defines if caching for options is active. The default value is `false`.

#### enable-caching-for-post

The `--enable-caching-for-post` option defines if caching for post is active.

#### enable-caching-string-sort

The `--enable-caching-string-sort` option defines if caching string sort is active.

#### file

> The `--file` option specifies the path to a JSON file containing the attributes of the Cache Settings configuration that will be updated. You can use `-` for reading from stdin.

Reviewer 2: ```suggestion
The `--file` option specifies the path to a JSON file containing the attributes of the Cache Settings configuration that'll be updated. You can use `-` for reading from stdin.
```
